core/src/main/java/nu/yona/server/goals/service/ActivityCategoryService.java

```diff
@@ -63,19 +65,35 @@
 		}
 	}
 
+	static final String CACHE_NAME = "activityCategories";
+
 	@Autowired
 	private ActivityCategoryRepository repository;
 
+	@Autowired(required = false)
+	private CacheConfiguration cacheConfiguration;
+
 	private static final Logger logger = LoggerFactory.getLogger(ActivityCategoryService.class);
 
+	@PostConstruct
+	public void register()
+	{
+		if (cacheConfiguration == null)
+		{
+			// Apparently running in a unit test that does not have all dependencies
+			return;
+		}
+		cacheConfiguration.registerCacheForMetrics(CACHE_NAME);
+	}
+
 	@Transactional
 	public ActivityCategoryDto getActivityCategory(UUID id)
 	{
 		ActivityCategory activityCategoryEntity = getEntityById(id);
 		return ActivityCategoryDto.createInstance(activityCategoryEntity);
 	}
 
-	@Cacheable(value = "activityCategorySet", key = "'instance'")
```

Comment:
This fixes a bug I assume?

Reply:
I'm not sure whether I would call it a bug, but the ``@CacheConfig`` annotation named the cache ``activityCategories`` while this annotation here named it ``activityCategorySet``, and the last one won. Now the cache is named ``activityCategories``.